portal/api-authentication.go

```diff
@@ -0,0 +1,242 @@
+// This file is part of MinIO Kubernetes Cloud
+// Copyright (c) 2019 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package portal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	jwtgo "github.com/dgrijalva/jwt-go"
+	jwtreq "github.com/dgrijalva/jwt-go/request"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// defaultJWTExpTime is the Portal expiration time for the authentication token.
+// Default is five minutes
+const defaultJWTExpTime = 5 * time.Minute
+
+var (
+	errAuthentication = errors.New("Authentication failed, check your access credentials")
+)
+
+// temporary user db for testing
+// {"userid":"hashedpassword",...}
+var users = map[string]map[string]map[string]string{
+	"acme": {
+		"cesnietor@acme.com": {
+			"password": "cesnietor_hashed",
+			"uuid":     "123e4567-e89b-12d3-a456-426655440000",
+		},
+		"daniel@acme.com": {
+			"password": "daniel_hashed",
+			"uuid":     "00112233-4455-6677-8899-aabbccddeeff",
+		},
+	},
+}
+
+// TODO: jwtKey should be gotten from a m3 global config
+var jwtKey = []byte("secret_key")
```

Comment:
I thought you were reading this from secrets

Reply:
yes, is not being used, already removed